x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/new_index/select_connector/select_connector.tsx

```diff
@@ -58,9 +55,14 @@ export const SelectConnector: React.FC = () => {
   const { isCloud } = useValues(KibanaLogic);
   const { hasPlatinumLicense } = useValues(LicensingLogic);
   const hasNativeAccess = isCloud;
-  const { service_type: serviceType } = parseQueryParams(search);
-  const [useNativeFilter, setUseNativeFilter] = useState(false);
-  const [useNonGAFilter, setUseNonGAFilter] = useState(true);
+  const { filter } = parseQueryParams(search);
+  const [selectedConnectorFilter, setSelectedConnectorFilter] = useState<string | null>(
+    Array.isArray(filter) ? filter[0] : filter ?? null
+  );
+  const useNativeFilter = selectedConnectorFilter === 'native';
```

Comment:
should probably use constants or a string union type here so we don't accidentally mess this up and the type system can help us when we inevitably typo the filter names.